erpnext/assets/doctype/asset/asset.py

```diff
@@ -255,9 +255,9 @@ def get_depreciation_amount(self, depreciable_value, total_number_of_depreciatio
 		precision = self.precision("gross_purchase_amount")
 
 		if row.depreciation_method in ("Straight Line", "Manual"):
+			depreciation_left = (cint(row.total_number_of_depreciations) - cint(self.number_of_depreciations_booked))
 			depreciation_amount = (flt(row.value_after_depreciation) -
-				flt(row.expected_value_after_useful_life)) / (cint(row.total_number_of_depreciations) -
-				cint(self.number_of_depreciations_booked))
+				flt(row.expected_value_after_useful_life)) / depreciation_left if depreciation_left else 1
 		else:
```

Comment:
@Anurag810 If no depreciations left, it should not take default 1. It should give a validation message that as all the depreciations has been booked, "value after depreciation" should be equal to "expected value after useful life"

Reply:
done